### PR TITLE
feat: support for downloading model files via rdma

### PR DESCRIPTION
**Pull request by 0xff-dev**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it
feat: support for downloading model files via rdma

#### Which issue(s) this PR fixes

Fix https://github.com/kubeagi/arcadia/issues/449

### CODE REVIEW

File: api/base/v1alpha1/datasource_types.go [hunk 1]
```diff
@@ -38,10 +38,12 @@ type DatasourceSpec struct {
 }
 
 type RDMA struct {
-	// Path on a model storage server, the usual storage path is /path/ns/mode-name, and the path field is /path/, which must end in /.
+	// ServerSavePath on a model storage server, the usual storage path is /path/ns/mode-name, and the path field is /path/, which must end in /.
 	// example: /opt/kubeagi/, /opt/, /
 	// +kubebuilder:validation:Pattern=(^\/$)|(^\/[a-zA-Z0-9\_.@-]+(\/[a-zA-Z0-9\_.@-]+)*\/$)
-	Path string `json:"path"`
+	ServerSavePath string `json:"path"`
```
Comment (bjwswang) [inline api/base/v1alpha1/datasource_types.go:44]: ServerPath string `json:"serverPath"`

File: pkg/worker/loader.go [hunk 2]
```diff
@@ -132,3 +132,54 @@ type LoaderGit struct{}
 func (loader *LoaderGit) Build(ctx context.Context, model *arcadiav1alpha1.TypedObjectReference) (any, error) {
 	return nil, ErrNotImplementedYet
 }
+
+var _ ModelLoader = (*RDMALoader)(nil)
+
+type RDMALoader struct {
```
Comment (bjwswang) [inline pkg/worker/loader.go:138]: Add a basic comment

File: pkg/worker/worker.go [hunk 3]
```diff
@@ -38,6 +38,11 @@ import (
 
 const (
 	WokerCommonSuffix = "-worker"
+
+	// TODO: Currently all nodes write the same savePath
+	RDMASavePath = "/raid10x12/rtranmodel"
```
Comment (bjwswang) [inline pkg/worker/worker.go:43]: Remove this line ?

File: api/base/v1alpha1/datasource_types.go (hunk 1 above)
Comment (0xff-dev) [reply api/base/v1alpha1/datasource_types.go:44]: How about `Path string`?
This path is the path of the sender on the one hand, and the path of the receiver on the other.

File: pkg/worker/worker.go [hunk 4]
```diff
@@ -38,6 +38,11 @@ import (
...
+
+	RDMALabel = "arcadia.kubeagi.k8s.com.cn/rdma"
```
Comment (bjwswang) [inline pkg/worker/worker.go:45]: RDMALabel -> RDMANodeLabel

File: pkg/worker/worker.go [hunk 5]
```diff
@@ -146,11 +151,24 @@ func NewPodWorker(ctx context.Context, c client.Client, s *runtime.Scheme, w *ar
 	podWorker.m = m
 
 	// default fields in a worker
-	storage := corev1.Volume{
-		Name: "models",
-		VolumeSource: corev1.VolumeSource{
-			EmptyDir: &corev1.EmptyDirVolumeSource{},
-		},
+	var storage corev1.Volume
+	if d.Spec.Type() != arcadiav1alpha1.DatasourceTypeRDMA {
```
Comment (bjwswang) [inline pkg/worker/worker.go:155]: use `switch case`  ?

File: pkg/worker/worker.go [hunk 6]
```diff
@@ -389,6 +419,33 @@ func (podWorker *PodWorker) Start(ctx context.Context) error {
 			Volumes:        []corev1.Volume{podWorker.storage},
 		},
 	}
+	if podWorker.storage.HostPath != nil {
```
Comment (bjwswang) [inline pkg/worker/worker.go:422]: Use datasourcetype for this check  ?

File: pkg/worker/worker.go [hunk 7]
```diff
@@ -372,6 +392,16 @@ func (podWorker *PodWorker) Start(ctx context.Context) error {
 	}
 	conRunner, _ := runner.(*corev1.Container)
 
+	if podWorker.storage.HostPath != nil {
+		conRunner.Lifecycle = &corev1.Lifecycle{
+			PreStop: &corev1.LifecycleHandler{
```
Comment (bjwswang) [inline pkg/worker/worker.go:397]: Nice!

File: pkg/worker/worker.go (hunk 6 above)
Comment (0xff-dev) [reply pkg/worker/worker.go:422]: The podWorker does not have a field for `datasource`, so if use datasource to judge it, we need to add the field.

File: deploy/charts/arcadia/crds/arcadia.kubeagi.k8s.com.cn_datasources.yaml [hunk 8]
```diff
@@ -103,10 +103,15 @@ spec:
                 description: RDMA configure RDMA pulls the model file directly from
                   the remote service to the host node.
                 properties:
+                  nodePaths:
+                    additionalProperties:
+                      type: string
```
Comment (bjwswang) [inline deploy/charts/arcadia/crds/arcadia.kubeagi.k8s.com.cn_datasources.yaml:108]: bump Chart Version

**Review by bjwswang — APPROVED** [review]

**Review by Abirdcfly — APPROVED** [review]

**Pull request merged**